orchagent/crmorch.cpp

```diff
@@ -0,0 +1,628 @@
+    {"acl_group_high_threshold", CrmResourceType::CRM_ACL_GROUP },
+    {"acl_entry_high_threshold", CrmResourceType::CRM_ACL_ENTRY },
+    {"acl_counter_high_threshold", CrmResourceType::CRM_ACL_COUNTER },
+    {"fdb_entry_high_threshold", CrmResourceType::CRM_FDB_ENTRY }
+};
+
+const map<string, CrmThresholdType> crmThreshTypeMap =
+{
+    { "percentage", CrmThresholdType::CRM_PERCENTAGE },
+    { "used", CrmThresholdType::CRM_USED },
+    { "free", CrmThresholdType::CRM_FREE }
+};
+
+const map<string, CrmResourceType> crmAvailCntsTableMap =
+{
+    { "crm_stats_ipv4_route_available", CrmResourceType::CRM_IPV4_ROUTE },
+    { "crm_stats_ipv6_route_available", CrmResourceType::CRM_IPV6_ROUTE },
+    { "crm_stats_ipv4_nexthop_available", CrmResourceType::CRM_IPV4_NEXTHOP },
+    { "crm_stats_ipv6_nexthop_available", CrmResourceType::CRM_IPV6_NEXTHOP },
+    { "crm_stats_ipv4_neighbor_available", CrmResourceType::CRM_IPV4_NEIGHBOR },
+    { "crm_stats_ipv6_neighbor_available", CrmResourceType::CRM_IPV6_NEIGHBOR },
+    { "crm_stats_nexthop_group_member_available", CrmResourceType::CRM_NEXTHOP_GROUP_MEMBER },
+    { "crm_stats_nexthop_group_object_available", CrmResourceType::CRM_NEXTHOP_GROUP_OBJECT },
+    { "crm_stats_acl_table_available", CrmResourceType::CRM_ACL_TABLE },
+    { "crm_stats_acl_group_available", CrmResourceType::CRM_ACL_GROUP },
+    { "crm_stats_acl_entry_available", CrmResourceType::CRM_ACL_ENTRY },
+    { "crm_stats_acl_counter_available", CrmResourceType::CRM_ACL_COUNTER },
+    { "crm_stats_fdb_entry_available", CrmResourceType::CRM_FDB_ENTRY }
+};
+
+const map<string, CrmResourceType> crmUsedCntsTableMap =
+{
+    { "crm_stats_ipv4_route_used", CrmResourceType::CRM_IPV4_ROUTE },
+    { "crm_stats_ipv6_route_used", CrmResourceType::CRM_IPV6_ROUTE },
+    { "crm_stats_ipv4_nexthop_used", CrmResourceType::CRM_IPV4_NEXTHOP },
+    { "crm_stats_ipv6_nexthop_used", CrmResourceType::CRM_IPV6_NEXTHOP },
+    { "crm_stats_ipv4_neighbor_used", CrmResourceType::CRM_IPV4_NEIGHBOR },
+    { "crm_stats_ipv6_neighbor_used", CrmResourceType::CRM_IPV6_NEIGHBOR },
+    { "crm_stats_nexthop_group_member_used", CrmResourceType::CRM_NEXTHOP_GROUP_MEMBER },
+    { "crm_stats_nexthop_group_object_used", CrmResourceType::CRM_NEXTHOP_GROUP_OBJECT },
+    { "crm_stats_acl_table_used", CrmResourceType::CRM_ACL_TABLE },
+    { "crm_stats_acl_group_used", CrmResourceType::CRM_ACL_GROUP },
+    { "crm_stats_acl_entry_used", CrmResourceType::CRM_ACL_ENTRY },
+    { "crm_stats_acl_counter_used", CrmResourceType::CRM_ACL_COUNTER },
+    { "crm_stats_fdb_entry_used", CrmResourceType::CRM_FDB_ENTRY }
+};
+
+CrmOrch::CrmOrch(DBConnector *db, string tableName):
+    Orch(db, tableName),
+    m_countersDb(new DBConnector(COUNTERS_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
+    m_countersCrmTable(new Table(m_countersDb.get(), COUNTERS_CRM_TABLE))
+{
+    SWSS_LOG_ENTER();
+
+    m_pollingInterval = chrono::minutes(CRM_POLLING_INTERVAL_DEFAULT);
```

Comment:
why not use seconds so that we have more granularity control, especially in the swss test, we can have much less time to finish the test.

Reply:
During design it was decided to use minutes, but I don't see any problem to change it to seconds.
So, should I change it?

Reply:
yes, please